packages/features/ee/workflows/components/agent-configuration/components/tabs/IncomingCallsTab.tsx

```diff
@@ -0,0 +1,190 @@
+        <div className="flex flex-col items-center space-y-6 text-center">
+          <div className="bg-muted flex h-16 w-16 items-center justify-center rounded-lg">
+            <Icon name="info" className="text-subtle h-8 w-8" />
+          </div>
+          <div className="space-y-2">
+            <h3 className="text-emphasis text-lg font-semibold">{t("setup_incoming_agent")}</h3>
+            <p className="text-subtle text-sm">{t("connect_a_phone_number_first")}</p>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  if (!inboundAgentId) {
+    return (
+      <div className="border-subtle rounded-xl border p-8">
+        <div className="flex flex-col items-center space-y-6 text-center">
+          <div className="bg-muted flex h-16 w-16 items-center justify-center rounded-lg">
+            <Icon name="phone-incoming" className="text-subtle h-8 w-8" />
+          </div>
+          <div className="space-y-2">
+            <h3 className="text-emphasis text-lg font-semibold">{t("setup_inbound_agent")}</h3>
+            <p className="text-subtle text-sm">{t("setup_inbound_agent_for_incoming_calls")}</p>
+          </div>
+          <Button
+            onClick={() => {
+              const phoneNumber = agentData?.outboundPhoneNumbers?.filter(
+                (phone) =>
+                  phone.subscriptionStatus === PhoneNumberSubscriptionStatus.ACTIVE ||
+                  !phone.subscriptionStatus
+              )[0]?.phoneNumber;
+
+              if (!phoneNumber) {
+                showToast(t("no_active_phone_number_available"), "error");
+                return;
+              }
+              if (!workflowStepId) {
+                showToast(t("workflow_step_not_configured"), "error");
+                return;
+              }
+
+              setupInboundAgentMutation.mutate({
+                phoneNumber,
+                teamId,
+                workflowStepId: workflowStepId,
+              });
+            }}
+            className="px-6"
+            loading={setupInboundAgentMutation.isPending}
+            disabled={setupInboundAgentMutation.isPending || !workflowStepId}>
+            {t("setup_inbound_agent")}
+          </Button>
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className="space-y-4">
+      <div>
+        <Label className="text-emphasis mb-1 block text-sm font-medium">{t("event_type")}</Label>
+        <p className="text-subtle mb-1.5 text-xs">{t("select_event_type_for_inbound_calls")}</p>
+        <Select
+          isSearchable={false}
+          innerClassNames={{ valueContainer: "font-medium" }}
+          className="text-sm font-medium"
+          isDisabled={readOnly || updateInboundAgentEventTypeMutation.isPending}
+          isLoading={updateInboundAgentEventTypeMutation.isPending}
+          onChange={(val) => {
+            if (val && inboundAgentId) {
+              updateInboundAgentEventTypeMutation.mutate({
```

Comment:
Why are we doing this `onChange` and not when clicking save ?

Reply:
Because eventType is not part of the form. This dropdown is just responsible for creating all the general tools for that event type `check_availability` and `book_appointment` and updating the prompt accordingly. This is much easier to debug and maintain.